### PR TITLE
add AD4 scoring function

### LINKED ISSUE #27: Can't find `unidock_ad4` after compiling from sources by `cmake`
Hi,

I am a bit confused that the README file said "Executable unidock supports vina and vinardo scoring functions, and unidock_ad4 supports ad4 scoring function."

But after I compiled from sources by `cmake`, only `unidock` exists not `unidock_ad4 `.

### PR DESCRIPTION
**Pull request by caic99**
fix #27

### CODE REVIEW

File: unidock/src/cuda/kernel.h [hunk 1]
```diff
@@ -45,7 +45,14 @@ void check(T result, char const *const func, const char *const file,
...
+
+// modified for vina1.2, should be larger, n*(n+1)/2, n=num_of_atom, select n=140
+#ifdef AD4
+#define MAX_P_DATA_M_DATA_SIZE 45150
+#else
+#define MAX_P_DATA_M_DATA_SIZE 11325 // vina
+#endif
+
 //#define MAX_NUM_OF_GRID_ATOMS 150
 #define FAST_SIZE 2051 // modified for vina1.2 m_max_cutoff^2 * factor + 3, ad4=13424
```
Comment (pkuyyj) [inline unidock/src/cuda/kernel.h:57]: Under ad4 scoring should be changed to 13424

**Comment by pkuyyj** [comment]
https://github.com/dptech-corp/Uni-Dock/blob/7c0cda06c94fc802194c55e7cec0cef1f834c31a/unidock/src/lib/scoring_function.h#L83C25-L83C25
This will affect FAST_SIZE

File: unidock/src/cuda/kernel.h (hunk 1 above)
Comment (caic99) [reply unidock/src/cuda/kernel.h:57]:
```suggestion

// modified for vina1.2 m_max_cutoff^2 * factor + 3
#ifdef AD4
#define FAST_SIZE 13424
#else
#define FAST_SIZE 2051
#endif

```

**Comment by caic99** [comment]
#27

**Pull request closed without merging**